src/modules/adaptors/Convex/ConvexCurveAdaptor.sol

```diff
@@ -0,0 +1,333 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { BaseAdaptor, ERC20, SafeTransferLib, Cellar, PriceRouter, Math } from "src/modules/adaptors/BaseAdaptor.sol";
+import { IBaseRewardPool } from "src/interfaces/external/Convex/IBaseRewardPool.sol";
+import { IBooster } from "src/interfaces/external/Convex/IBooster.sol";
+import { CurvePool } from "src/interfaces/external/Curve/CurvePool.sol";
+import { console } from "@forge-std/Test.sol";
+
+/**
+ * @title Convex-Curve Platform Adaptor
+ * @dev This adaptor is specifically for Convex contracts.
+ * @notice Allows cellars to have positions where they are supplying, staking LPTs, and claiming rewards to Convex markets.
+ * @author crispymangoes, 0xEinCodes
+ * @dev this may not work for Convex with other protocols / platforms / networks. It is important to keep these associated to Curve-Convex Platform on Mainnet
+ * TODO: refine adaptorData now that there is bytes4 with it too.
+ */
+contract ConvexCurveAdaptor is BaseAdaptor {
+    using SafeTransferLib for ERC20;
+    using Math for uint256;
+
+    struct PoolInfo {
```

Comment:
Is this needed anymore?

Reply:
Oh this is used in `assetsUsed`. Can it be moved to an interface file?

Reply:
I believe it can, I have removed it from this and it is already part of the `IBooster.sol` interface file. I just realized it is not needed though since we use a getter to access the tuple from the `booster` contract.